azure-kusto-data/azure/kusto/data/client.py

```diff
@@ -853,19 +853,61 @@ def execute_streaming_ingest(
 
         self._execute(endpoint, database, None, stream, self._streaming_ingest_default_timeout, properties)
 
-    def _execute(
-        self, endpoint: str, database: str, query: Optional[str], payload: Optional[io.IOBase], timeout: timedelta, properties: ClientRequestProperties = None
+    def execute_streaming_query(
+        self, database: str, query: str, timeout: timedelta = _KustoClientBase._query_default_timeout, properties: Optional[ClientRequestProperties] = None
+    ) -> Response:

```

Comment:
I don't understand the result - in the non-streaming query we return a wrapper of type "KustoResponseDataSet" to make it easy to use the response as tables and provide type conversions.
I think the raw result will be hard to use unless I miss something here.

Reply:
This is modeled after the java api - https://github.com/Azure/azure-kusto-java/pull/170

To convert it to KustoResponseDataSet we need all of the data in-memory, which of course isn't possible for a huge amount of data.
The usecase is for them to dump it into a file and parse the json there.

@yihezkel correct me if I'm wrong

Reply:
I don't think its useful. 
I did some work to parse the result in lazy manner, maybe we can find the gap between what we have today and full lazy evaluation and make everything "streaming"

Reply:
The Java SDK was a special case. As discussed, I recommend using streaming object creation to create a KustoResponseDataSet incrementally. I'm not familiar with Python support for this concept, but hopefully we wou;dn't need to change the datatype to support this.